packages/instantsearch.js/src/widgets/related-products/related-products.tsx

```diff
@@ -0,0 +1,222 @@
+/** @jsx h */
+
+import { createRelatedProductsComponent } from 'instantsearch-ui-components';
+import { Fragment, h, render } from 'preact';
+
+import TemplateComponent from '../../components/Template/Template';
+import connectRelatedProducts from '../../connectors/related-products/connectRelatedProducts';
+import { prepareTemplateProps } from '../../lib/templating';
+import {
+  getContainerNode,
+  createDocumentationMessageGenerator,
+} from '../../lib/utils';
+
+import type {
+  RelatedProductsWidgetDescription,
+  RelatedProductsConnectorParams,
+  RelatedProductsRenderState,
+} from '../../connectors/related-products/connectRelatedProducts';
+import type { PreparedTemplateProps } from '../../lib/templating';
+import type { Template, WidgetFactory, Hit, Renderer } from '../../types';
+import type { RecommendResultItem } from 'algoliasearch-helper';
+import type {
+  RecommendClassNames,
+  RelatedProductsProps as RelatedProductsUiProps,
+} from 'instantsearch-ui-components';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'related-products',
+});
+
+const RelatedProducts = createRelatedProductsComponent({
+  createElement: h,
+  Fragment,
+});
+
+type CreateRendererProps = {
+  containerNode: HTMLElement;
+  cssClasses: RelatedProductsCSSClasses;
+  renderState: {
+    templateProps?: PreparedTemplateProps<Required<RelatedProductsTemplates>>;
```

Comment:
I think this doesn't need `Required`, I know `Hits` has this but other widgets don't. That way we don't have an issue with ESLint line 60/61